text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+  });
+```
+
+__NOTE:__
+
+* Provisioned mode for write capacity could only be used with autoscaling. There is no way to provision
+an exact write capacity unit in global tables.
+* Write capacity could not be configured on a per replica basis. You can only modify read capacity per
+replica.
+* You can either define `readCapacity` or `readWithAutoScaling` but not both at the same time.
+
+#### Global Secondary Indexes
+
+You can add global secondary indexes to your global table. These would be the same for each replica of the
+table. There is some configuration that could be specified per replica.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    sortKey: { name: 'Bar', type: AttributeType.STRING },
+    globalSecondaryIndex: [
+      {
+        indexName: 'FooTableGsi',
+        partitionKey: { name: 'Bar', type: AttributeType.STRING },
+      },
+    ],
+  });
+```
+
+You can add global secondary index with `addGlobalSecondaryIndex` method.
+
+```typescript
+  globalTable.addGlobalSecondaryIndex({
+    indexName: 'FooTableGsi',
+    partitionKey: { name: 'Bar', type: AttributeType.STRING },
+  });
+```
+
+__NOTE:__
+
+* You can create up to 20 global secondary indexes.
+* You can only create or delete one global secondary index in a single stack operation.
+
+#### Local Secondary Indexes
+
+You can add local secondary indexes to your global table. These would be the same for each replica of
+the table.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    sortKey: { name: 'Bar', type: AttributeType.STRING },
+    localSecondaryIndex: [
+      {
+        indexName: 'FooTableLsi',
+        sortKey: { name: 'Foo', type: AttributeType.STRING },
+      },
+    ],
+  });
+```
+
+You can add local secondary index with `addLocalSecondaryIndex` method.
+
+```typescript
+  globalTable.addLocalSecondaryIndex({
+    indexName: 'FooTableLsi',
+    sortKey: { name: 'Foo', type: AttributeType.STRING },
+  });
+```
+
+__NOTE:__
+
+* You need a sort key to define a local secondary index.
+* You can create up to five local secondary indexes.
+
+#### Encryption
+
+There are three types of encryptions that are available for global tables.
+
+1. `AWS_OWNED`: This uses a KMS key for encryption that is owned by DynamoDB. This is the default
+for global tables.
+2. `AWS_MANAGED`: A KMS key is created in your account and is managed by AWS.
+3. `CUSTOMER_MANAGED`: You can specify a KMS key that is managed by you for encrytion.
+
+This would be the same for each replica.
+
+```typescript
+  new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    encryption: TableEncryption.AWS_MANAGED,
+  });
+```
+
+__NOTE__:
+
+* If the encryption mode is CUSTOMER_MANAGED, then you would need to specify a KMS key for each replica.
```

Comment:
A different KMS key for each region for each replica? What does this look like in practice?

I think cross-stack references will become a problem here, no?